PyP100/PyP100.py

```diff
@@ -220,6 +220,69 @@ def setBrightness(self, brightness):
 			errorCode = ast.literal_eval(decryptedResponse)["error_code"]
 			errorMessage = self.errorCodes[str(errorCode)]
 
+	def setColourTemp(self, colourtemp):
+		URL = f"http://{self.ipAddress}/app?token={self.token}"
+		Payload = {
+			"method": "set_device_info",
+			"params":{
+				"color_temp": colourtemp
+			},
+			"requestTimeMils": int(round(time.time() * 1000)),
+		}
+
+		headers = {
+			"Cookie": self.cookie
+		}
+
+		EncryptedPayload = self.tpLinkCipher.encrypt(json.dumps(Payload))
+
+		SecurePassthroughPayload = {
+			"method": "securePassthrough",
+			"params":{
+				"request": EncryptedPayload
+			}
+		}
+
+		r = requests.post(URL, json=SecurePassthroughPayload, headers=headers)
+
+		decryptedResponse = self.tpLinkCipher.decrypt(r.json()["result"]["response"])
+
+		if ast.literal_eval(decryptedResponse)["error_code"] != 0:
+			errorCode = ast.literal_eval(decryptedResponse)["error_code"]
+			errorMessage = self.errorCodes[str(errorCode)]
+
+	def setColour(self, hue, saturation):
```

Comment:
```suggestion
	def setColour(self, hue, saturation, brightness=None):
```
You can set brightness at the same time as hue and saturation, so it could be an optional parameter here.

Reply:
Or maybe they should all be separate, one for brightness, one for hue, one for saturation?

Reply:
You can't send hue without saturation, or vice versa. Gives error -1008.